src/System.Net.Sockets/src/System/Net/Sockets/IOControlKeepAlive.Windows.cs

```diff
@@ -30,6 +30,15 @@ public static bool IsNeeded
             }
         }
 
+        public static void Get(SafeSocketHandle handle, SocketOptionName optionName, byte[] optionValue, ref int optionLength)
+        {
+            Debug.Assert(optionValue != null);
+            Debug.Assert(optionValue.Length == sizeof(int));
+
+            BitConverter.TryWriteBytes(optionValue.AsSpan(), Get(handle, optionName));
```

Comment:
This should throw a SocketException if TryWriteBytes returns false, right?  It'd be good to have a test for that case, too... that's at least what the documentation leads me to believe.

Reply:
Or, actually, it seems like this should in that case return a failed SocketError, and then the code above the PAL layer will throw appropriately.

Reply:
`SocketError.Fault` needs to be used since it is what generates an exception for:
```csharp
GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, 0);

byte[] bufferNull = null;
GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferNull);

byte[] bufferLenghtZero = new byte[0];
GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferLenghtZero);

byte[] bufferShort = new byte[1];
GetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveBuffer, bufferShort);
```